src/Microsoft.Data/DataFrame.IDataView.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+
+namespace Microsoft.Data
+{
+    public partial class DataFrame : IDataView
+    {        
+        // TODO: support shuffling
+        bool IDataView.CanShuffle => false;
+
+        private DataViewSchema _schema;
+        DataViewSchema IDataView.Schema
+        {
+            get
+            {
+                if (_schema != null && _schema.Count == ColumnCount)
+                {
+                    return _schema;
```

Comment:
Maybe set this to null in the `InsertColumn`, `SetColumn` and `RemoveColumn` methods? Presumably, someone can run an ML algo=>change the dataframe(thereby changing the schema)=>run ML again and so on.

Reply:
Good call.  I'll invalidate this when the DataFrame changes.